pip_audit/_service/osv.py

```diff
@@ -90,11 +90,7 @@ def query(self, spec: Dependency) -> Tuple[Dependency, List[VulnerabilityResult]
 
             id = vuln["id"]
 
-            # The summary is intended to be shorter, so we prefer it over
-            # details, if present. However, neither is required.
             description = vuln.get("summary")
-            if description is None:
-                description = vuln.get("details")
```

Comment:
I kind of wonder if we should fall back on a portion of the description, maybe everything up until the first newline or something?

Reply:
That'll work in many cases, although there'll be a handful where it'll produce something relatively useless (like a Markdown `## header`). Overall probably worth it, so I'll add it.

Reply:
Actually, maybe we should just fall back on `details`, but with newlines replaced with spaces? That'll result in a big blob of text in the worst case, but it'll avoid the situation where we render the first line and it doesn't happen to be useful.

Reply:
I think that's a good idea.

Reply:
That's much nicer 🙂 


Name | Version | ID | Fix Versions | Description
--- | --- | --- | --- | ---
twisted | 18.9.0 | PYSEC-2019-128 | 19.2.1 | In Twisted before 19.2.1, twisted.web did not validate or sanitize URIs or HTTP methods, allowing an attacker to inject invalid characters such as CRLF.
twisted | 18.9.0 | PYSEC-2020-260 | 20.3.0rc1 | In Twisted Web through 19.10.0, there was an HTTP request splitting vulnerability. When presented with a content-length and a chunked encoding header, the content-length took precedence and the remainder of the request body was interpreted as a pipelined request.
twisted | 18.9.0 | PYSEC-2019-129 | 19.7.0rc1 | In words.protocols.jabber.xmlstream in Twisted through 19.2.1, XMPP support did not verify certificates when used with TLS, allowing an attacker to MITM connections.
twisted | 18.9.0 | PYSEC-2020-259 | 20.3.0rc1 | In Twisted Web through 19.10.0, there was an HTTP request splitting vulnerability. When presented with two content-length headers, it ignored the first header. When the second content-length value was set to zero, the request body was interpreted as a pipelined request.
twisted | 18.9.0 | PYSEC-2022-27 | 22.1.0 | twisted is an event-driven networking engine written in Python. In affected versions twisted exposes cookies and authorization headers when following cross-origin redirects. This issue is present in the `twited.web.RedirectAgent` and `twisted.web. BrowserLikeRedirectAgent` functions. Users are advised to upgrade. There are no known workarounds.
twisted | 18.9.0 | PYSEC-2022-195 | 22.4.0 | Twisted is an event-based framework for internet applications, supporting Python 3.6+. Prior to version 22.4.0rc1, the Twisted Web HTTP 1.1 server, located in the `twisted.web.http` module, parsed several HTTP request constructs more leniently than permitted by RFC 7230. This non-conformant parsing can lead to desync if requests pass through multiple HTTP parsers, potentially resulting in HTTP request smuggling. Users who may be affected use Twisted Web's HTTP 1.1 server and/or proxy and also pass requests through a different HTTP server and/or proxy. The Twisted Web client is not affected. The HTTP 2.0 server uses a different parser, so it is not affected. The issue has been addressed in Twisted 22.4.0rc1. Two workarounds are available: Ensure any vulnerabilities in upstream proxies have been addressed, such as by upgrading them; or filter malformed requests by other means, such as configuration of an upstream proxy.
twisted | 18.9.0 | GHSA-32gv-6cf3-wcmq | 19.10.0 | ### Impact Twisted web servers that utilize the optional HTTP/2 support suffer from the following flow-control related vulnerabilities:  Ping flood: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9512 Reset flood: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9514 Settings flood: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-9515  A Twisted web server supports HTTP/2 requests if you've installed the [`http2` optional dependency set](https://twistedmatrix.com/documents/19.2.0/installation/howto/optional.html).  ### Workarounds There are no workarounds.  ### References https://github.com/Netflix/security-bulletins/blob/master/advisories/third-party/2019-002.md  ### For more information If you have any questions or comments about this advisory: * Open an issue in [Twisted's Trac](https://twistedmatrix.com/trac/)